light/provider/http/http.go

```diff
@@ -100,10 +100,16 @@ func NewWithClientAndOptions(chainID string, client rpcclient.RemoteClient, opti
 	}
 }
 
-func (p *http) String() string {
+// Identifies the provider with an IP in string format
+func (p *http) ID() string {
```

Comment:
I'm not sure why we need two methods for this. I think we should either use `ID` (which has the appeal of being self-describing) or replace its use entirely with `String`.